code/game/machinery/Sleeper.dm

```diff
@@ -145,12 +162,56 @@
 
 /obj/machinery/sleeper/New()
 	..()
+	RefreshParts()
 	spawn( 5 )
+		update_icon()
 		if(orient == "RIGHT")
-			icon_state = "sleeper_0-r"
+			generate_console(get_step(get_turf(src), WEST))
+		else
+			generate_console(get_step(get_turf(src), EAST))
 		return
 	return
 
+/obj/machinery/sleeper/Destroy()
+	..()
+	qdel(connected)
+
+/obj/machinery/sleeper/update_icon()
+	if(occupant)

```

Comment:
Same as the other update_icon you can construct these

icon_state = "sleeper_[occupant ? "1" : "0"][orient == "LEFT" ? null : "-r"]"